### PR TITLE
[NFC][Autotune] Use device_prop.regsPerMultiprocessor instead of hardcoded reg number.

### PR DESCRIPTION
**Pull request by htyu**
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/csrc/cuda/Module.cpp [hunk 1]
```diff
@@ -892,6 +892,11 @@ static void registerCudaDeviceProperties(PyObject* module) {
       .def_readonly(
           "max_threads_per_multi_processor",
           &cudaDeviceProp::maxThreadsPerMultiProcessor)
+#if !USE_ROCM
+      // NVIDA only property
+      .def_readonly(
+          "regsPerMultiprocessor", &cudaDeviceProp::regsPerMultiprocessor)
```
Comment (jansel) [inline torch/csrc/cuda/Module.cpp:898]:
```suggestion
          "regs_per_multiprocessor", &cudaDeviceProp::regsPerMultiprocessor)
```
to match the style above

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_inductor/triton_heuristics.py [hunk 2]
```diff
@@ -209,6 +209,8 @@ def precompile(self, warm_cache_only_with_cc=None):
                 config.dynamic_scale_rblock
                 and self.heuristic_type == HeuristicType.REDUCTION
                 and self.size_hints is not None
+                # Disable for AMDGPU as Triton is not ready to return n_regs for a compiled_binary.
```
[the remark this answers is not in the document]
Comment (htyu) [reply torch/_inductor/triton_heuristics.py:212]: According to what I observed, on AMDGPU, Triton always return 0 for nreg and nspill. This sounds something we should fix on the Triton side.

```
[2023-11-14 16:36:16,157] torch._inductor.triton_heuristics: [DEBUG] COORDESC: XBLOCK: 1, RBLOCK: 2048, num_warps: 8, num_ctas: 1, num_stages: 1, enable_warp_specialization: False, enable_persistent: False: 0.012640, nreg 0, nspill 0, #shared-mem 2048

[2023-11-14 16:36:17,249] torch._inductor.triton_heuristics: [DEBUG] COORDESC: XBLOCK: 1, RBLOCK: 1024, num_warps: 8, num_ctas: 1, num_stages: 1, enable_warp_specialization: False, enable_persistent: False: 0.012480, nreg 0, nspill 0, #shared-mem 2048
```

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**